### PR TITLE
Dsl1 2.5.0

### LINKED ISSUE #1017: DSL1: Input file name collision when merging two already merged BAMs in `additional_library_merge` step.
## Check Documentation

I have checked the following places for your error:

- [x] [nf-core website: troubleshooting](https://nf-co.re/usage/troubleshooting)
- [x] [nf-core/eager pipeline documentation](https://nf-co.re/nf-core/eager/usage)
      - nf-core/eager FAQ/troubleshooting can be found [here](https://nf-co.re/eager/usage#troubleshooting-and-faqs)

## Description of the bug
In niche cases where multiple UDG treatments exist for a sample, and multiple libraries have each of these treatments, a file name collision kills the pipeline at the `additional_library_merge` step.

## Steps to reproduce

Steps to reproduce the behaviour:

1. Command line: `nextflow run ...` (any input that requires merging of two already merged library-level BAMs during `additional_library_merge` step.
2. See error: 

```
Caused by:
  Process `additional_library_merge` input file name collision -- There are multiple input files for each of the following file names: MIS010_ss_libmerged.trimmed.bam, MIS010_ss_libmerged.trimmed.bam.bai
```

## Expected behaviour

The BAMs initial libmerged should have unique names, to avoid such errors.

## Log files

Have you provided the following extra information/files:

- [ ] The command used to run the pipeline
- [ ] The `.nextflow.log` file 
- [x] The exact error: 

## System

- Hardware: HPC
- Executor: sge 
- OS: Ubuntu
- Version 20.04.6 LTS 

## Nextflow Installation

- Version:  21.10.6 build 5660

## Container engine

- Engine: Singularity
- version: singularity version 3.7.1
- Image tag: 2.4.5

### LINKED ISSUE #1020: DSL1: Add possibility to use MapDamage for damage estimation
## Is your feature request related to a problem? Please describe

DamageProfiler does not have an option to limit the amount of reads used for damage estimation. With really large BAM files, the amount of time and memory required to get a damage estimate becomes unrealistic, when the estimate is likely quite accurate already from 10k reads onwards.

## Describe the solution you'd like

It would be good to include MapDamage2 as an alternative option for damage calculation, since it includes an option to limit the number of reads used. Current implementation of MapDamage2 in nf-core/eager only runs if the user has requested BAM rescaling. Would be nice to have a quick way of getting damage plots without needing to do the time-consuming rescaling step.
Results form the stat estimation can also be provided to the rescaling step to speed that up if a user decides to do both. 

## Describe alternatives you've considered

It is possible to use PMDtools as an alternative atm, but that is slower than either alternative, and can fail unexpectedly at times.

## Additional context

https://github.com/Integrative-Transcriptomics/DamageProfiler/issues/59
https://github.com/Integrative-Transcriptomics/DamageProfiler/issues/58
Sadly, even if these were to be fixed, newer versions of Damageprofiler cannot be used because of Java requirement clashes with `gatk 3.5`, so fixing these issues there would only fix the behaviour of the pipeline from `3.0+`

### PR DESCRIPTION
**Pull request by TCLamnidis**
- Added option to use MapDamage instead of DamageProfiler for damage estimation. Also added options to downsample to a specified number of reads for damage calculation with mapDamage. 
- Fixed a niche error causing file name collisons.

Closes #1020.
Closes #1017.
Clsoes #1031. 

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
    - [x] If you've added a new tool - add to the software_versions process and a regex to `scrape_software_versions.py`
    - [x] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/>nf-core/eager/tree/master/.github/CONTRIBUTING.md)
    - [x] If necessary, also make a PR on the nf-core/eager _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint .`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: docs/output.md
Comment (jfy133) [inline docs/output.md:501]:
```suggestion
DamageProfiler and mapDamage are tools that calculate a variety of standard 'aDNA' metrics from a BAM file. The primary plots here are the misincorporation and length distribution plots. Ancient DNA undergoes depurination and hydrolysis, causing fragmentation of molecules into gradually shorter fragments, and cytosine to thymine deamination damage, that occur on the subsequent single-stranded overhangs at the ends of molecules.
```

File: main.nf
Comment (jfy133) [inline main.nf:200]: mother of god :laughing:

File: main.nf
Comment (jfy133) [inline main.nf:2111]: Maybe `mapdamage_profiling` would be clearer? But not important

**Review by jfy133 — APPROVED** [review]
Needs adding of mapDamage to README list of tools but otherwise LGTM

File: main.nf
Comment (TCLamnidis) [reply main.nf:200]: DSL1, baby!

File: main.nf
Comment (TCLamnidis) [reply main.nf:2111]: the section is called Damage Calculation. so could change to that. tho i doubt it matters much

Comment (TCLamnidis) [inline main.nf:2111]:
```suggestion
process mapdamage_calculation {
```

**Comment by TCLamnidis** [comment]
Surprised citation isn't there already given it is already used for damage rescaling 😬

**Comment by TCLamnidis** [comment]
Added mapDamage to Default Pipeline Steps in README. Standardised tool name to `mapDamage` across all docs (as the name is in the GH repository of the tool)

**Comment by TCLamnidis** [comment]
Will wait for mapdamage module to be added to multiqc, then should be ok to merge

**Comment by jfy133** [comment]
Should add a big fat warning top of the readme sayi g maximum NDC aversion of 22.10.6

**Comment by TCLamnidis** [comment]
Regarding the Bowtie read groups, the suggested edits would also need to be propagated to the bwa job.
Specifically: 
 - Changing the RG-ID to the sample name from the library name, and 
  - adding a separate field to the RG with the library ID

File: docs/output.md
Comment (jfy133) [inline docs/output.md:529]:
```suggestion
> **NB:** An important difference to note compared to the mapDamage tool, which DamageProfiler is otherwise an exact-re-implementation of, is that the percent frequency on the Y axis is not fixed between 0 and 0.3, and will 'zoom' into small values the less damage there is
```

**Review by jfy133 — APPROVED** [review]
mapdamage stuff looks good, lets wait for feedback on the RG

**Comment by TCLamnidis** [comment]
TODO: 
MultiQC does not pick up the mapdamage outputs locally, even though they are correctly published within the work directory and the module is activated. Have to check what is wrong.

**Comment by TCLamnidis** [comment]
I think that's everything!

**Review by jfy133 — APPROVED** [review]
LGTM assuming old linting working and tests pass

**Comment by TCLamnidis** [comment]
@ltcrod Please have a look at the RG information for the mappers. Let me know if anything would be better implemented differently.

**Comment by TCLamnidis** [comment]
Linting works with nf-core v1.14. only failures are about template matching.

**Review by ltcrod — APPROVED** [review]

**Pull request merged**
